lib/src/cli.rs

```diff
@@ -384,17 +384,31 @@ async fn upgrade(opts: UpgradeOpts) -> Result<()> {
         crate::status::get_status_require_booted(sysroot)?;
     let imgref = host.spec.image.as_ref();
     // If there's no specified image, let's be nice and check if the booted system is using rpm-ostree
-    if imgref.is_none()
-        && host
+    if imgref.is_none() {
+        if host
             .status
             .booted
             .as_ref()
             .map_or(false, |b| b.incompatible)
-    {
-        return Err(anyhow::anyhow!(
-            "Booted deployment contains local rpm-ostree modifications; cannot upgrade via bootc. You can run `rpm-ostree reset` to undo the modifications."
-        ));
+        {
+            return Err(anyhow::anyhow!(
+                "Booted deployment contains local rpm-ostree modifications; cannot upgrade via bootc. You can run `rpm-ostree reset` to undo the modifications."
+            ));
+        }
+
+        if host.status.staged.is_some()
+            && host
+                .status
+                .staged
+                .as_ref()
+                .map_or(false, |b| b.incompatible)
```

Comment:
But for a bigger cleanup, I think we can merge the logic for checking something like this:

```
let booted_incompatible = host
            .status
            .booted
            .as_ref()
            .map_or(false, |b| b.incompatible);
let staged_incompatible = host
                .status
                .staged
                .as_ref()
                .map_or(false, |b| b.incompatible);
if booted_incompatible || staged_incompatible {
   return Err...
}
```
or so?

Reply:
sounds good, updated